docs/reference/developer/adr/0016-do-not-support-attrs.rst

```diff
@@ -0,0 +1,118 @@
+ADR 0016: Do not support "attrs"
+================================
+
+- Status: draft
+- Deciders:
+- Date: 2023-05-16
+
+Context
+-------
+
+Background
+~~~~~~~~~~
+
+Arbitrary user-defined attributes appear in many applications.
+Examples include:
+
+- HDF5 attrs for datasets and groups
+- Attributes in NetCDF
+- Attributes in Zarr
+- Attributes in NeXus (typically using HDF5 attrs)
+- ``xarray.Variable.attrs`` and ``xarray.DataArray.attrs``
+
+Scipp's ``DataArray`` supports ``attrs``.
+Note that currently ``attrs`` doubles as a place to store "unaligned" coords, which is not directly related to the topic discussed here.
+However, ``scipp.Variable`` does not support ``attrs``.
+In part this is due to a reduced need for ``attrs``, since Scipp has native support for physical units, which is one of the recurring applications of ``attrs`` in other libraries.
+The other reason is legacy:
+``scipp.Variable`` was written in C++ and was intended to be lightweight and simple.
+
+When interacting with files such as HDF5 or NeXus we encounter a recurring need to load arbitrary attributes.
+Currently this is not possible in Scipp, at least not if we want to represent an HDF5 dataset as a ``scipp.Variable``.
+Support for ``scipp.Variable.attrs`` would allow for representing arbitrary attributes in a ``scipp.Variable``.
+The question is whether this is a good idea.
+
+Analysis
+~~~~~~~~
+
+Addition of ``scipp.Variable.attrs`` would add a significant amount complexity to the code base.
+It would likely also impact performance in some applications, as we would need to handle references to Python objects (requiring GIL acquisition in C++ code) or shared pointers.
+It could furthermore increase the chance of running into problems with Python garbage collection similar to the existing (low-propability) issues described in https://github.com/scipp/scipp/issues/2813.
```

Comment:
```suggestion
It could furthermore increase the chance of running into problems with Python garbage collection similar to the existing (low-probability) issues described in https://github.com/scipp/scipp/issues/2813.
```